plugins/module_utils/_api/utils/socket.py

```diff
@@ -43,7 +42,7 @@ def read(socket, n=4096):
     recoverable_errors = (errno.EINTR, errno.EDEADLK, errno.EWOULDBLOCK)
 
     if PY3 and not isinstance(socket, NpipeSocket):
-        if sys.platform == 'win32':
+        if not hasattr(select, "poll"):
```

Comment:
I don't really understand the specifics of what's going on here, but the conditional is quite different. Was the issue actually that `select.poll` was not defined on Windows platforms? If the attribute exists on Windows but did not work (or didn't work correctly for this situation) then this test would cause failures.

I guess, this code being vendored, it never will run on Windows (unless I ever get around to trying to get Ansible Python modules to execute on Windows but alas....).

Still, what was the motivation for this change?

Reply:
This has been cherry-picked from https://github.com/docker/docker-py/commit/78439ebbe1aae77ff0fd9b666894d80807182e28 to support [eventlet](https://github.com/eventlet/eventlet). Our code neither supports Windows (outside WSL2) nor does it support eventlet, I'd still like the socket code to be as close as possible to Docker SDK for Python's code even for features we have no need for.